internal/functions/Test-ShouldProcess.ps1

```diff
@@ -0,0 +1,9 @@
+function Test-ShouldProcess {
```

Comment:
The reason for this was to be able to mock the result in Pester, which allowed me to do more unit testing.

https://github.com/pester/Pester/issues/560

Reply:
Understood..then this needs @niphlod @FriedrichWeinmann @potatoqualitee review to approve as if it is done in this command it should be used across the board.

Reply:
Just a learning note @awickham10 for workflow it is best to start this type of thing by filing an issue (not sure if you did and I just missed it). This allows us all to discuss and plan proper path being that this will be a chance across the whole module. We can hash out everything and prevent you from spinning and waiting on a PR to get merged.

Reply:
Understood, thanks. I was actually needed it for work, so went ahead and started in on it. Should I open up an issue now, or doesn't matter?

Reply:
No need to file an issue, I'll discuss it with the others directly.
We can bulk-replace it for most of the module (did a quick try-out - 730 out of 740 uses can be bulk-updated without further inspection. Question is whether we want to, will try to get back to here on that soonish.

Reply:
We had that discussion and unanimously agree: This idea is neat!
Given that, this internal function is totally fine the way and not an issue PR-wise.

Merge once the rest of this PR is cleared, I'll update it a bit and then do a bulk update across the entire module